kratos/sources/model_part.cpp

```diff
@@ -1438,9 +1459,20 @@ ModelPart&  ModelPart::CreateSubModelPart(std::string const& NewSubModelPartName
         //KRATOS_ERROR << "There is an already existing sub model part with name \"" << NewSubModelPartName << "\" in model part: \"" << Name() << "\"" << std::endl;
     }
 
-void ModelPart::AddSubModelPart(Kratos::shared_ptr<ModelPart> pThisSubModelPart)
-{
-   KRATOS_ERROR << "cannot add a submodelpart, since submodelparts are univocally owned by their father " << std::endl;
+KRATOS_DEPRECATED void ModelPart::AddSubModelPart(ModelPart& rThisSubModelPart)
+{
+    KRATOS_ERROR << "cannot add a submodelpart, since submodelparts are univocally owned by their father " << std::endl;
+//     if (mSubModelParts.find(pThisSubModelPart->Name()) != mSubModelParts.end())
```

Comment:
If is deprecated I would send a warning not an error

Reply:
Is not important for them if the model is the wrong one. They never use it. If we want to maintain the legacy we should maintain it until the date we are promising.